ubiconfig/_impl/loaders/gitlab.py

```diff
@@ -47,20 +47,24 @@ def load_all(self, recursive=False):
 
         return ubi_configs
 
-    def _pre_load(self, per_page):
+    def _pre_load(self):
         """Iterate all branches to get a mapping of {file_path: branch,...}
         """
         files_branch_map = {}
         branches = self._get_branches()
         LOG.info("Loading config files from all branches: %s", branches)
         for branch in branches:
-            file_list_api = self._repo_api.get_file_list_api(branch=branch,
-                                                             per_page=per_page)
-            json_response = self._session.get(file_list_api).json()
-            file_list = [file['path'] for file in json_response
-                         if file['name'].endswith(('.yaml', '.yml'))]
-            for file in file_list:
-                files_branch_map[file] = branch
+            headers_api = self._repo_api.get_file_list_api(branch=branch)
+            headers = self._session.get(headers_api).headers
```

Comment:
Minor: looks like the first page will be requested twice, which is a bit wasteful.

I'd probably have written it something like this to avoid having to do one request before the loop just to get page count:

```python
  page = 1
  while True:
    response = ... # get current page
    file_list = ... # do stuff with response
    if page >= (response.headers.get('X-Total-Pages') or 1):
      # got the last page, done
      break
    # there's more pages, keep looping
    page += 1
```

Reply:
+1 or instead of GET just do a HEAD to get only headers.